### PR TITLE
Update default plugin configs

### PR DESCRIPTION
**Pull request by SharkyKZ**
Pull Request for Issue # .

### Summary of Changes
As per #20563, this syncs default plugin param values in installation SQL files with changes made in PRs #20544, #20456, #20422, #20412 and #20410. This mainly affects numeric values that were stored as strings.

### Testing Instructions
Code review.

### Documentation Changes Required
No.

### CODE REVIEW

File: installation/sql/mysql/joomla.sql [hunk 1]
```diff
@@ -565,38 +565,38 @@ INSERT INTO `#__extensions` (`extension_id`, `package_id`, `name`, `type`, `elem
...
+(407, 0, 'plg_content_pagebreak', 'plugin', 'pagebreak', 'content', 0, 1, 1, 0, '', '{"title":1,"multipage_toc":1,"showall":1}', '', '', 0, '0000-00-00 00:00:00', 4, 0),
+(408, 0, 'plg_content_pagenavigation', 'plugin', 'pagenavigation', 'content', 0, 1, 1, 0, '', '{"position":1}', '', '', 0, '0000-00-00 00:00:00', 5, 0),
 (409, 0, 'plg_content_vote', 'plugin', 'vote', 'content', 0, 0, 1, 0, '', '', '', '', 0, '0000-00-00 00:00:00', 6, 0),
-(410, 0, 'plg_editors_codemirror', 'plugin', 'codemirror', 'editors', 0, 1, 1, 1, '', '{"lineNumbers":"1","lineWrapping":"1","matchTags":"1","matchBrackets":"1","marker-gutter":"1","autoCloseTags":"1","autoCloseBrackets":"1","autoFocus":"1","theme":"default","tabmode":"indent"}', '', '', 0, '0000-00-00 00:00:00', 1, 0),
+(410, 0, 'plg_editors_codemirror', 'plugin', 'codemirror', 'editors', 0, 1, 1, 1, '', '{"lineNumbers":1,"lineWrapping":1,"matchTags":1,"matchBrackets":1,"marker-gutter":1,"autoCloseTags":1,"autoCloseBrackets":1,"autoFocus":1,"theme":"default","tabmode":"indent"}', '', '', 0, '0000-00-00 00:00:00', 1, 0),
```
Comment (Quy) [inline installation/sql/mysql/joomla.sql]: Default should be 0 to match in XML:
lineNumbers
lineWrapping
marker-gutter (should be markerGutter)

No settings:
autoFocus
tabmode

Comment (SharkyKZ) [reply installation/sql/mysql/joomla.sql]: Existing SQL values should take priority over XML. Based on current values, all options should be enabled on new installations. I'll update #20544.

File: installation/sql/mysql/joomla.sql [hunk 2]
```diff
@@ -565,38 +565,38 @@ INSERT INTO `#__extensions` (`extension_id`, `package_id`, `name`, `type`, `elem
...
 (416, 0, 'plg_editors-xtd_readmore', 'plugin', 'readmore', 'editors-xtd', 0, 1, 1, 0, '', '', '', '', 0, '0000-00-00 00:00:00', 4, 0),
-(417, 0, 'plg_search_categories', 'plugin', 'categories', 'search', 0, 1, 1, 0, '', '{"search_limit":"50","search_content":"1","search_archived":"1"}', '', '', 0, '0000-00-00 00:00:00', 0, 0),
-(418, 0, 'plg_search_contacts', 'plugin', 'contacts', 'search', 0, 1, 1, 0, '', '{"search_limit":"50","search_content":"1","search_archived":"1"}', '', '', 0, '0000-00-00 00:00:00', 0, 0),
-(419, 0, 'plg_search_content', 'plugin', 'content', 'search', 0, 1, 1, 0, '', '{"search_limit":"50","search_content":"1","search_archived":"1"}', '', '', 0, '0000-00-00 00:00:00', 0, 0),
-(420, 0, 'plg_search_newsfeeds', 'plugin', 'newsfeeds', 'search', 0, 1, 1, 0, '', '{"search_limit":"50","search_content":"1","search_archived":"1"}', '', '', 0, '0000-00-00 00:00:00', 0, 0),
+(417, 0, 'plg_search_categories', 'plugin', 'categories', 'search', 0, 1, 1, 0, '', '{"search_limit":50,"search_content":1,"search_archived":1}', '', '', 0, '0000-00-00 00:00:00', 0, 0),
```
Comment (Quy) [inline installation/sql/mysql/joomla.sql:594]: In XML files, `search_content` and `search_archived` are 0 except in `plg_search_content`

File: installation/sql/mysql/joomla.sql [hunk 3]
```diff
@@ -565,38 +565,38 @@ INSERT INTO `#__extensions` (`extension_id`, `package_id`, `name`, `type`, `elem
...
+(433, 0, 'plg_user_profile', 'plugin', 'profile', 'user', 0, 0, 1, 0, '', '{"register-require_address1":1,"register-require_address2":1,"register-require_city":1,"register-require_region":1,"register-require_country":1,"register-require_postal_code":1,"register-require_phone":1,"register-require_website":1,"register-require_favoritebook":1,"register-require_aboutme":1,"register-require_tos":1,"register-require_dob":1,"profile-require_address1":1,"profile-require_address2":1,"profile-require_city":1,"profile-require_region":1,"profile-require_country":1,"profile-require_postal_code":1,"profile-require_phone":1,"profile-require_website":1,"profile-require_favoritebook":1,"profile-require_aboutme":1,"profile-require_tos":1,"profile-require_dob":1}', '', '', 0, '0000-00-00 00:00:00', 0, 0),
```
Comment (Quy) [inline installation/sql/mysql/joomla.sql]: register-require_tos is 0 in XML

File: installation/sql/mysql/joomla.sql [hunk 4]
```diff
@@ -613,7 +613,7 @@ INSERT INTO `#__extensions` (`extension_id`, `package_id`, `name`, `type`, `elem
 (448, 0, 'plg_twofactorauth_totp', 'plugin', 'totp', 'twofactorauth', 0, 0, 1, 0, '', '', '', '', 0, '0000-00-00 00:00:00', 0, 0),
 (449, 0, 'plg_authentication_cookie', 'plugin', 'cookie', 'authentication', 0, 1, 1, 0, '', '', '', '', 0, '0000-00-00 00:00:00', 0, 0),
 (450, 0, 'plg_twofactorauth_yubikey', 'plugin', 'yubikey', 'twofactorauth', 0, 0, 1, 0, '', '', '', '', 0, '0000-00-00 00:00:00', 0, 0),
-(451, 0, 'plg_search_tags', 'plugin', 'tags', 'search', 0, 1, 1, 0, '', '{"search_limit":"50","show_tagged_items":"1"}', '', '', 0, '0000-00-00 00:00:00', 0, 0),
+(451, 0, 'plg_search_tags', 'plugin', 'tags', 'search', 0, 1, 1, 0, '', '{"search_limit":50,"show_tagged_items":1}', '', '', 0, '0000-00-00 00:00:00', 0, 0),
```
Comment (Quy) [inline installation/sql/mysql/joomla.sql:626]: show_tagged_items is 0 in XML

File: installation/sql/mysql/joomla.sql (hunk 2 above)
Comment (SharkyKZ) [reply installation/sql/mysql/joomla.sql:594]: Updated values in plugins.

File: installation/sql/mysql/joomla.sql (hunk 4 above)
Comment (SharkyKZ) [reply installation/sql/mysql/joomla.sql:626]: Updated value in plugin.

File: plugins/user/profile/profile.php [hunk 5]
```diff
@@ -273,7 +273,7 @@ public function onContentPrepareForm(JForm $form, $data)
 		}
 
 		$tosArticle = $this->params->get('register_tos_article');
-		$tosEnabled = $this->params->get('register-require_tos', 0);
+		$tosEnabled = ($this->params->get('register-require_tos', 0) == 2);
```
Comment (Quy) [inline plugins/user/profile/profile.php]: Remove outside `()` which are not necessary.

File: installation/sql/mysql/joomla.sql [hunk 6]
```diff
@@ -565,38 +565,38 @@ INSERT INTO `#__extensions` (`extension_id`, `package_id`, `name`, `type`, `elem
...
-(408, 0, 'plg_content_pagenavigation', 'plugin', 'pagenavigation', 'content', 0, 1, 1, 0, '', '{"position":"1"}', '', '', 0, '0000-00-00 00:00:00', 5, 0),
+(407, 0, 'plg_content_pagebreak', 'plugin', 'pagebreak', 'content', 0, 1, 1, 0, '', '{"title":1,"multipage_toc":1,"showall":1}', '', '', 0, '0000-00-00 00:00:00', 4, 0),
+(408, 0, 'plg_content_pagenavigation', 'plugin', 'pagenavigation', 'content', 0, 1, 1, 0, '', '{"position":1}', '', '', 0, '0000-00-00 00:00:00', 5, 0),
 (409, 0, 'plg_content_vote', 'plugin', 'vote', 'content', 0, 0, 1, 0, '', '', '', '', 0, '0000-00-00 00:00:00', 6, 0),
-(410, 0, 'plg_editors_codemirror', 'plugin', 'codemirror', 'editors', 0, 1, 1, 1, '', '{"lineNumbers":"1","lineWrapping":"1","matchTags":"1","matchBrackets":"1","marker-gutter":"1","autoCloseTags":"1","autoCloseBrackets":"1","autoFocus":"1","theme":"default","tabmode":"indent"}', '', '', 0, '0000-00-00 00:00:00', 1, 0),
+(410, 0, 'plg_editors_codemirror', 'plugin', 'codemirror', 'editors', 0, 1, 1, 1, '', '{"lineNumbers":1,"lineWrapping":1,"matchTags":1,"matchBrackets":1,"markerGutter":1,"autoCloseTags":1,"autoCloseBrackets":1,"theme":"default"}', '', '', 0, '0000-00-00 00:00:00', 1, 0),
```
Comment (Quy) [inline installation/sql/mysql/joomla.sql]: Please confirm if `"theme":"default"` should be empty. https://github.com/joomla/joomla-cms/pull/20544/files#diff-56697cf5f4f7b09581b782ea92afc869L226

Comment (SharkyKZ) [reply installation/sql/mysql/joomla.sql]: Yes, it should be empty.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on db777e4bd73af64de114b8624c58deeedab0d662<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20893">issues.joomla.org/tracker/joomla-cms/20893</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on db777e4bd73af64de114b8624c58deeedab0d662<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20893">issues.joomla.org/tracker/joomla-cms/20893</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20893">issues.joomla.org/tracker/joomla-cms/20893</a>.</sub>

**Comment by richard67** [comment]
@SharkyKZ It seems meanwhile this PR got conflicts because changes in staging meanwhile. Do you think you can continue this PR or make a new one? Or is this PR obsolete now because things are right meanwhile in staging?

**Comment by HLeithner** [comment]
@SharkyKZ could you please resolve the conflicts?

**Comment by SharkyKZ** [comment]
Conflicts solved.

**Comment by HLeithner** [comment]
Do we have tests for all 3 database  engines?

**Comment by HLeithner** [comment]
I removed RTC.

We need tests for mysql, pgsql and mssql please. Include the version in the testing comment.

Thanks for your effort testing this.

**Comment by HLeithner** [comment]
3rd database test pending....

**Comment by zero-24** [comment]
Dear @SharkyKZ 

in preperation of the upcomming release of Joomla 3.10 we have used GitHubs rename feature to rename the staging branch into 3.10-dev. Usually GitHub moves all existing PRs towards the new branch just fine, but here it didnt work. The reason seems to be that the fork of the CMS that was used as base for this PR has been deleted so GitHub does no longer have a base to rebase the PR against the new branch and we are also not able to reopen the PR. For that reason GitHub closed this PR in my name, when this issue is still valid It would require a new PR against the new 3.10-dev or 4.0-dev branch.

**Pull request closed without merging**